src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowManager.java

```diff
@@ -101,12 +103,20 @@ public Future<Void> updatePiece(Piece pieceToUpdate, boolean createItem, boolean
         }
         return null;
       })
-      .onSuccess(v -> promise.complete())
       .onFailure(t -> {
         logger.error("User to update piece with id={}", holder.getPieceToUpdate().getId(), t.getCause());
-        promise.fail(t);
-      });
-    return promise.future();
+        var param1 = new Parameter().withKey("pieceId").withValue(pieceToUpdate.getId());
+        var param2 = new Parameter().withKey("poLineId").withValue(pieceToUpdate.getPoLineId());
+        var causeParam = new Parameter().withKey("cause").withValue(t.getMessage());
+        if (StringUtils.isNotBlank(t.getMessage()) && t.getMessage().contains("Barcode must be unique")) {
+          logger.error("The barcode associate with item '{}' is not unique, it cannot be updated", param1.getValue(), t);
+          throw new HttpException(500, BARCODE_IS_NOT_UNIQUE, List.of(param1, param2, causeParam));
+        } else {
+          logger.error("Item associated with piece '{}' cannot be updated", param1.getValue(), t);
+          throw new HttpException(500, ITEM_UPDATE_FAILED, List.of(param1, param2, causeParam));
```

Comment:
we can not use ITEM_UPDATE_FAILED because we can fail for any reason not connected with item, for example editing po lines

Reply:
As a solution we can remain here only if without else, or move logic to appropriate place after updating Item. Second option is preferable